pkg/minishift/update/update_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+)
+
+var (
+	archivePath string
+	err         error
+	port        int
+)
+
+func TestDownloadAndVerifyArchive(t *testing.T) {
+	port = randomNum(1000, 9999)
```

Comment:
Why not use the same approach we use already for _github.go_ - https://github.com/minishift/minishift/blob/435f9bc64b8d1054b8ddd045c411e7a3a24beccc/pkg/testing/http_mock.go#L53-L53

Instead of starting a file server we modify the HTTPRoundTripper to return something. This way, we don't have to use a custom HTTP server and we also don't run the risk of port conflicts. In your code we can get random errors, if a port is selected which is already occupied. This will be heaps of fun to debug.